spec/manifest_pr_checker_spec.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+require_relative 'spec_helper'
+
+module Danger
+  describe Danger::ManifestPRChecker do
+    it 'is a plugin' do
+      expect(described_class.new(nil)).to be_a Danger::Plugin
+    end
+
+    describe 'with Dangerfile' do
+      before do
+        @dangerfile = testing_dangerfile
+        @plugin = @dangerfile.manifest_pr_checker
+      end
+
+      describe 'Bundler' do
+        it 'returns a warning when a PR changed the Gemfile but not the Gemfile.lock' do
+          modified_files = ['Gemfile']
+          allow(@plugin.git).to receive(:modified_files).and_return(modified_files)
+
+          @plugin.check_gemfile_lock_updated
+
+          expected_warning = ['Gemfile was changed without updating Gemfile.lock. Please run `bundle install` or `bundle update <updated_gem>`.']
+          expect(@dangerfile.status_report[:warnings]).to eq expected_warning
+        end
+
+        it 'returns no warnings when both the Gemfile and the Gemfile.lock were updated' do
+          modified_files = ['Gemfile', 'Gemfile.lock']
+          allow(@plugin.git).to receive(:modified_files).and_return(modified_files)
+
+          @plugin.check_gemfile_lock_updated
+
+          expect(@dangerfile.status_report[:warnings]).to be_empty
+        end
+
+        it 'returns no warnings when only the Gemfile.lock was updated' do
+          modified_files = ['Gemfile.lock']
+          allow(@plugin.git).to receive(:modified_files).and_return(modified_files)
+
+          @plugin.check_gemfile_lock_updated
+
+          expect(@dangerfile.status_report[:warnings]).to be_empty
+        end
+      end
+
+      describe 'CocoaPods' do
+        it 'returns a warning when a PR changed the Podfile but not the Podfile.lock' do
+          modified_files = ['Podfile']
+          allow(@plugin.git).to receive(:modified_files).and_return(modified_files)
+
+          @plugin.check_podfile_lock_updated
+
+          expected_warning = ['Podfile was changed without updating Podfile.lock. Please run `bundle exec pod install`.']
+          expect(@dangerfile.status_report[:warnings]).to eq expected_warning
+        end
+
+        it 'returns no warnings when both the Podfile and the Podfile.lock were updated' do
+          modified_files = ['Podfile', 'Podfile.lock']
+          allow(@plugin.git).to receive(:modified_files).and_return(modified_files)
+
+          @plugin.check_podfile_lock_updated
+
+          expect(@dangerfile.status_report[:warnings]).to be_empty
+        end
+
+        it 'returns a warning when a PR changed a custom located Podfile but not the corresponding Podfile.lock' do
```

Comment:
We need another test that tests cases like:

```
modified_files = ['./dir1/Podfile', './dir2/Podfile', './dir1/Podfile.lock']
```
To ensure that while the one in `dir1` found its corresponding lockfile, the one in `dir2` missing its lockfile update is still correctly flagged by the plugin.

Reply:
btw, I'd suggest to use TDD here (aka Red-Green-Refactoring), i.e.:
 - First implement the additional tests for the expected behavior,
 - Then run `rspec` and see the tests go red, proving that the tests you wrote would properly catch such a regression
 - Then update the implementation (e.g. [see if my suggested snippet above](https://github.com/Automattic/dangermattic/pull/3#discussion_r1265804027) would work) to make the test go green
 - Then once the test is green, optionally refactor to optimize the code… if needed.

Reply:
I've implemented the changes we discussed on Monday, by the way.